dtfs-central-api/src/repositories/tfm-facilities-repo/tfm-facilities.repo.ts

```diff
@@ -296,4 +308,35 @@ export class TfmFacilitiesRepo {
     });
     return numberOfFoundDocuments > 0;
   }
+
+  public static async upsertPortalFacilityAmendmentDraft(amendment: PortalFacilityAmendment, auditDetails: AuditDetails): Promise<UpdateResult> {
+    const collection = await this.getCollection();
+
+    const findFilter: Filter<TfmFacility> = {
+      _id: { $eq: new ObjectId(amendment.facilityId) },
+      'facilitySnapshot.dealId': { $eq: new ObjectId(amendment.dealId) },
+    };
+
+    const removeDraftAmendmentsFilter: UpdateFilter<TfmFacility> = {
+      $pull: {
+        amendments: { type: AMENDMENT_TYPES.PORTAL, status: { $ne: AMENDMENT_STATUS.COMPLETED } },
```

Comment:
Do we want to remove amendments if the status is `IN_PROGRESS`?

Reply:
Thanks @MarRobSoftwire  - can we add some documentation about this please?

There's a lot going on in this function

Reply:
Added :)